Bottom: 00f466efa1a8832981dfcebee031e824a6c103ed
Top:    0306c4a8f653d9713685909e385da5b3df2339dd
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-12 02:50:19 +0100

Refresh of speedup_poly_poly_checking

---

diff --git a/src/find.c b/src/find.c
index 7b6507f..a1b7167 100644
--- a/src/find.c
+++ b/src/find.c
@@ -1814,6 +1814,7 @@ LookupLOConnectionsToArc (ArcTypePtr Arc, Cardinal LayerGroup)
             return True;
 
           /* now check all polygons */
+          printf ("Slow pour path for arcs\n");
           POUR_LOOP (LAYER_PTR (layer));
           {
             POURPOLYGON_LOOP (pour);
@@ -1957,6 +1958,7 @@ LookupLOConnectionsToLine (LineTypePtr Line, Cardinal LayerGroup,
           /* now check all polygons */
           if (PolysTo)
             {
+              printf ("Slow pour path for lines\n");
               POUR_LOOP (LAYER_PTR (layer));
               {
                 POURPOLYGON_LOOP (pour);
@@ -2056,6 +2058,7 @@ LOTouchesLine (LineTypePtr Line, Cardinal LayerGroup)
             return (True);
 
           /* now check all polygons */
+          printf ("Slow pour path for lines (LOTouchesLine)\n");
           POUR_LOOP (LAYER_PTR (layer));
           {
             POURPOLYGON_LOOP (pour);
@@ -2443,9 +2446,9 @@ static int
 LOCtoPolyPourPolygon_callback (const BoxType * b, void *cl)
 {
   PourTypePtr pour = (PourTypePtr) b;
-  struct pv_info *i = (struct pv_info *) cl;
+  struct lo_info *i = (struct pv_info *) cl;
 
-  return r_search (pour->polygon_tree, (BoxType *) &i->pv,
+  return r_search (pour->polygon_tree, (BoxType *) &i->polygon,
                    NULL, LOCtoPolyPolygon_callback, i);
 }
